File: plugin/src/android/LivenessFacePlugin.java
package com.example.outsystems.livenessface;

import org.apache.cordova.CallbackContext;
import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.PluginResult;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class LivenessFacePlugin extends CordovaPlugin {

    @Override
    public boolean execute(String action, JSONArray args, final CallbackContext callbackContext) {
        switch (action) {
            case "initialize":
                cordova.getThreadPool().execute(() -> handleInitialize(args, callbackContext));
                return true;
            case "startDetection":
                cordova.getThreadPool().execute(() -> handleStartDetection(args, callbackContext));
                return true;
            default:
                callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.INVALID_ACTION));
                return false;
        }
    }

    private void handleInitialize(JSONArray args, CallbackContext callbackContext) {
        try {
            JSONObject options = args.optJSONObject(0);
            // TODO: initialize native SDK with options
            callbackContext.success();
        } catch (Exception e) {
            callbackContext.error(e.getMessage());
        }
    }

    private void handleStartDetection(JSONArray args, CallbackContext callbackContext) {
        try {
            JSONObject options = args.optJSONObject(0);
            // TODO: invoke native liveness detection SDK
            JSONObject result = new JSONObject();
            result.put("status", "not_implemented");
            callbackContext.success(result);
        } catch (JSONException e) {
            callbackContext.error(e.getMessage());
        }
    }
}

File: plugin/src/ios/LivenessFacePlugin.swift
import Foundation
import Cordova

@objc(LivenessFace)
class LivenessFacePlugin: CDVPlugin {
    @objc(initialize:)
    func initialize(command: CDVInvokedUrlCommand) {
        // TODO: Configure native SDK using command.arguments.first
        let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK)
        commandDelegate.send(pluginResult, callbackId: command.callbackId)
    }

    @objc(startDetection:)
    func startDetection(command: CDVInvokedUrlCommand) {
        // TODO: Launch liveness detection workflow and return payload
        let payload: [String: Any] = ["status": "not_implemented"]
        let pluginResult = CDVPluginResult(status: CDVCommandStatus_OK, messageAs: payload)
        commandDelegate.send(pluginResult, callbackId: command.callbackId)
    }
}
